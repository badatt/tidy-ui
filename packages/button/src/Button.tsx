import React from 'react';
import { styled, css } from '@tidy-ui/theme';
import { RotatingCircleIcon } from '@tidy-ui/icons';
import { Size, Variant, Status, Appearance } from '@tidy-ui/types';
import { sizeStyles } from './style';

interface IButtonProps extends React.HTMLAttributes<HTMLButtonElement> {
  size?: Size;
  kind?: Variant | Status;
  appearance?: Appearance;
  disabled?: boolean;
  loading?: boolean;
  text?: string;
  icon?: JSX.Element;
  fullWidth?: boolean;
  uppercase?: boolean;
}

const basic = css<IButtonProps>`
  ${({ theme, kind }) => css`
    color: ${theme.palette[kind || 'neutral'].main};
    background-color: #e2e8f0;
    border: 1px solid transparent;
    box-shadow: rgba(0, 0, 0, 0) 0px 0px 0px 0px, rgba(0, 0, 0, 0) 0px 0px 0px 0px,
      rgba(0, 0, 0, 0.1) 0px 10px 15px -3px, rgba(0, 0, 0, 0.1) 0px 4px 6px -4px;
  `}
`;

const filled = css<IButtonProps>`
  ${({ theme, kind }) => css`
    color: ${theme.palette[kind || 'neutral'].contrastText};
    background-color: ${theme.palette[kind || 'neutral'].main};
    border: 1px solid transparent;
    box-shadow: rgba(0, 0, 0, 0) 0px 0px 0px 0px, rgba(0, 0, 0, 0) 0px 0px 0px 0px,
      ${theme.palette[kind || 'neutral'].light} 0px 10px 15px -3px,
      ${theme.palette[kind || 'neutral'].light} 0px 4px 6px -4px;
  `}
`;

const outlined = css<IButtonProps>`
  ${({ theme, kind }) => css`
    color: ${theme.palette[kind || 'neutral'].main};
    background-color: transparent;
    border: 1px solid ${theme.palette[kind || 'neutral'].light};
  `}
`;

const hero = css<IButtonProps>`
  ${({ theme, kind }) => css`
    padding: 1rem 1.5rem;
    border-radius: 0.75rem;
    font-size: 3rem;
    line-height: 2.125rem;
    color: ${theme.palette[kind || 'neutral'].contrastText};
    background-color: ${theme.palette[kind || 'neutral'].main};
    border: 1px solid transparent;
    width: 100%;
    text-align: center;
    justify-content: center;
    box-shadow: rgba(0, 0, 0, 0) 0px 0px 0px 0px, rgba(0, 0, 0, 0) 0px 0px 0px 0px,
      ${theme.palette[kind || 'neutral'].light} 0px 10px 15px -3px,
      ${theme.palette[kind || 'neutral'].light} 0px 4px 6px -4px;
  `}
`;

const ButtonRoot = styled.button<IButtonProps>`
  display: flex;
  justify-content: center;
  align-items: center;
  ${({ size, disabled, fullWidth, appearance, uppercase }) => css`
    padding: ${sizeStyles[size || 'md'].padding};
    border-radius: ${sizeStyles[size || 'md'].borderRadius};
    font-size: ${sizeStyles[size || 'md'].fontSize};
    line-height: ${sizeStyles[size || 'md'].lineHeight};
    ${disabled &&
    css`
      cursor: not-allowed;
      opacity: 0.5;
    `}
    ${fullWidth &&
    css`
      width: 100%;
      text-align: center;
      justify-content: center;
    `}
    ${uppercase &&
    css`
      text-transform: uppercase;
    `}
    ${appearance === 'basic' && basic}
    ${appearance === 'filled' && filled}
    ${appearance === 'outlined' && outlined}
    ${appearance === 'hero' && hero}
  `}
`;

const Icon = styled.i<IButtonProps>`
  display: flex;
  justify-content: center;
  align-items: center;
  ${({ size, text }) => css`
    height: ${sizeStyles[size || 'md'].iconSize};
    width: ${sizeStyles[size || 'md'].iconSize};
    ${text === ''
      ? css`
          margin-right: 0em;
        `
      : css`
          margin-right: 0.25em;
        `};
  `}
`;

const Button = React.forwardRef<HTMLButtonElement, IButtonProps>((props, ref) => {
  const { className, children, loading, icon, text, disabled, ...rest } = props;
  return (
    <ButtonRoot role="button" {...rest} disabled={disabled || loading}>
      {icon && <Icon {...props}>{icon}</Icon>}
      {loading && (
        <Icon {...props}>
          <RotatingCircleIcon />
        </Icon>
      )}
      {text}
      {children}
    </ButtonRoot>
  );
});

Button.defaultProps = {
  disabled: false,
  loading: false,
  text: '',
  fullWidth: false,
  uppercase: false,
};

export { Button };
