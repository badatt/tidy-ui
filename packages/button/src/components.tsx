import { applyStandardOverrideStyles, css, hsla, Icon, styled } from '@tidy-ui/commons';
import { focusVisibility, hero, iconOnly, outlined, primary, simple, sizeStyles } from './styles';
import { IAnchorProps, IButtonClusterProps, IButtonGroupProps, IButtonProps, IIconButtonProps } from './types';

/**
 * Internal root component with styles
 *
 * @internal
 */
const AnchorRoot = styled.a<IAnchorProps>`
  ${({ theme: { palette, isDark }, disabled, girth, tone }) => css`
    margin: 0 0.5ch;
    display: inline-block;
    text-decoration: none;
    padding: 0.2em 0.3em;
    line-height: 1.5em;
    font-size: calc(${sizeStyles[girth!].fontSize} + 0.125rem);
    color: ${isDark ? palette[tone!][400] : palette[tone!][600]};
    ${disabled
      ? css`
          cursor: not-allowed;
          pointer-events: none;
          opacity: 0.6;
        `
      : css`
          background-image: linear-gradient(
            ${hsla(palette[tone!].shades[500], 0.7)},
            ${hsla(palette[tone!].shades[500], 0.7)}
          );
          background-size: 100% 0em;
          background-position-y: 100%;
          background-repeat: no-repeat;
          transition: background-size 0.2s ease-in-out;
          :hover,
          :active,
          :focus {
            background-size: 100% 25%;
          }
        `}
  `}
  ${applyStandardOverrideStyles}
`;

const AnchorWithIcon = styled.span<IAnchorProps>`
  display: flex;
  align-items: baseline;
  gap: 0.25ch;

  & svg,
  & span {
    height: 0.8em !important;
    width: 0.8em !important;
  }
`;

/**
 * Internal root component with styles
 *
 * @internal
 */
const ButtonRoot = styled.button<IButtonProps>`
  cursor: pointer;
  display: flex;
  justify-content: center;
  align-items: center;
  user-select: none;
  white-space: nowrap;
  text-decoration: none;
  text-align: center;
  vertical-align: middle;
  transition: all 200ms linear;
  ${({ girth, disabled, isStretched, variant, icon, isUppercase }) => css`
    font-size: ${sizeStyles[girth!].fontSize};
    padding: 0.2em 0.3em;
    border-radius: 0.3em;
    line-height: 1.5em;
    ${disabled &&
    css`
      cursor: not-allowed;
      opacity: 0.6;
      transition: none;
    `}
    ${isStretched &&
    css`
      width: 100%;
      text-align: center;
      justify-content: center;
    `}
    ${isUppercase &&
    css`
      text-transform: uppercase;
    `}
    ${variant === 'simple' && !icon && simple}
    ${variant === 'primary' && !icon && primary}
    ${variant === 'outlined' && !icon && outlined}
    ${variant === 'hero' && !icon && hero}
    ${icon && iconOnly}
    ${focusVisibility}
  `}
  ${applyStandardOverrideStyles}
`;

/**
 * Internal root component with styles
 *
 * @internal
 */
const ButtonGroupRoot = styled.div<IButtonGroupProps>`
  display: flex;
  ${({ isVertical, isStretched, isUnified }) => css`
    ${isVertical
      ? css`
          flex-direction: column;
          & > :first-child {
            border-bottom-left-radius: 0 !important;
            border-bottom-right-radius: 0 !important;
            ${isUnified &&
            css`
              border-bottom: 0 !important;
            `}
          }
          & > :last-child {
            border-top-left-radius: 0 !important;
            border-top-right-radius: 0 !important;
            ${isUnified &&
            css`
              border-top: 0 !important;
            `}
          }
          & > :not(:first-child):not(:last-child) {
            ${isUnified &&
            css`
              border-top: 0 !important;
              border-bottom: 0 !important;
            `}
          }
        `
      : css`
          & > :first-child {
            border-top-right-radius: 0 !important;
            border-bottom-right-radius: 0 !important;
            ${isUnified &&
            css`
              border-right: 0 !important;
            `}
          }
          & > :last-child {
            border-top-left-radius: 0 !important;
            border-bottom-left-radius: 0 !important;
            ${isUnified &&
            css`
              border-left: 0 !important;
            `}
          }
          & > :not(:first-child):not(:last-child) {
            ${isUnified &&
            css`
              border-right: 0 !important;
              border-left: 0 !important;
            `}
          }
        `}
    ${isStretched &&
    css`
      width: 100% !important;
      & * {
        flex-grow: 1 !important;
        flex-shrink: 1 !important;
        flex-basis: 1% !important;
      }
    `}
    & * {
      box-shadow: none !important;
    }
    & > :not(:first-child):not(:last-child) {
      border-radius: 0 !important;
    }
  `}
  ${applyStandardOverrideStyles}
`;

/**
 * Internal root component with styles
 *
 * @internal
 */
const ButtonClusterRoot = styled.div<IButtonClusterProps>`
  display: flex;
  flex-wrap: wrap;
  gap: 0.5rem;
  & * {
    box-shadow: none !important;
    &:hover {
      box-shadow: none !important;
    }
  }
  ${({ isStretched }) => css`
    ${isStretched &&
    css`
      width: 100% !important;
      & * {
        flex-grow: 1 !important;
        flex-shrink: 1 !important;
        flex-basis: 1% !important;
      }
    `}
  `}
  ${applyStandardOverrideStyles}
`;

/**
 * Internal root component with styles
 *
 * @internal
 */
const IconButtonRoot = styled.button<IIconButtonProps>`
  display: flex;
  justify-content: center;
  align-items: center;
  ${({ girth, disabled, isIconOnly, variant, placement }) => css`
    font-size: ${sizeStyles[girth!].fontSize};
    ${isIconOnly
      ? css`
          padding: 0.3em;
          border-radius: 50%;
        `
      : css`
          padding: 0.2rem;
          border-radius: 0.4em;
        `}
    ${disabled &&
    css`
      cursor: not-allowed;
      opacity: 0.5;
    `}
    ${placement == 'right' &&
    css`
      flex-direction: row-reverse;
    `}
    ${variant === 'primary' && primary}
    ${variant === 'outlined' && outlined}
  `}
  ${applyStandardOverrideStyles}
`;

/**
 * The Wrapper component for Icon
 *
 * @internal
 */
const IconButtonIconWrapper = styled.span<Partial<IIconButtonProps>>`
  padding: 0.2em;
`;

/**
 * The Wrapper component for isIconOnly variant buttons
 *
 * @internal
 */
const IconButtonIconOnlyWrapper = styled.span<Partial<IIconButtonProps>>`
  ${({ isIconOnly }) => css`
    ${isIconOnly
      ? css`
          padding: 0.2em;
        `
      : css`
          padding: 0.4em 0.6em;
        `}
  `}
`;

/**
 * Internal Icon with styles
 *
 * @internal
 */
const IconButtonIcon = styled.span<Partial<IIconButtonProps>>`
  display: flex;
  justify-content: center;
  align-items: center;
  ${({ isIconOnly }) => css`
    ${isIconOnly
      ? css`
          height: 1.4em;
          width: 1.4em;
        `
      : css`
          height: 1.2em;
          width: 1.2em;
        `}
  `}
`;

/**
 * Internal component for children other than Icon
 *
 * @internal
 */
const IconButtonChildWrapper = styled.span<Partial<IIconButtonProps>>`
  display: inline-flex;
  justify-content: center;
  align-items: center;
  vertical-align: middle;
  white-space: nowrap;
  user-select: none;
  ${({ placement }) => css`
    ${placement == 'right'
      ? css`
          padding-left: 0.4em;
        `
      : css`
          padding-right: 0.4em;
        `}
  `}
`;

const LoadingIcon = styled(Icon.RotatingCircle)`
  height: 1.5em;
  width: 1.5em;
  margin: 0 0.25em 0 0;
`;

export {
  AnchorRoot,
  AnchorWithIcon,
  ButtonClusterRoot,
  ButtonGroupRoot,
  ButtonRoot,
  IconButtonChildWrapper,
  IconButtonIcon,
  IconButtonIconOnlyWrapper,
  IconButtonIconWrapper,
  IconButtonRoot,
  LoadingIcon,
};
