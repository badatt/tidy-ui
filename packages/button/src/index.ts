export * from './Button';
export * from './Anchor';
