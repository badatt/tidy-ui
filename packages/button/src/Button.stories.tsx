import React from "react";

import { Button } from "./Button";

export default {
  component: Button,
  title: "Design System/Button",
};

export const basic = () => <Button />;
