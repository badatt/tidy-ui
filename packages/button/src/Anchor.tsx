import React, { forwardRef } from 'react';
import { AnchorIcon, AnchorRoot } from './components';
import { IAnchorProps } from './types';

/**
 * Anchor component is used to represent hyperlinks on a page, a good substitute for default
 * html `a` tag
 *
 */
const Anchor = forwardRef<HTMLAnchorElement, IAnchorProps>((props, ref) => {
  const { children, launch, ...rest } = props;
  return (
    <AnchorRoot role="link" ref={ref} {...rest} target={launch ? '_blank' : '_top'}>
      {children}
      {launch && <AnchorIcon />}
    </AnchorRoot>
  );
});

Anchor.defaultProps = {
  disabled: false,
  launch: false,
};

export { Anchor };
