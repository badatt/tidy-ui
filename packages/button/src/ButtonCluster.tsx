import React from 'react';
import { ButtonClusterRoot } from './components';
import { IButtonClusterProps } from './types';

/**
 * ButtonCluster can group several Buttons & ButtonGroups to bring in that
 * toolbar effect, that you usually see in embedded editors
 *
 */
const ButtonCluster = React.forwardRef<HTMLDivElement, IButtonClusterProps>((props, ref) => {
  const { children, disabled, ...rest } = props;
  return (
    <ButtonClusterRoot role="button" ref={ref} {...rest}>
      {React.Children.map(children, (c) => {
        const child = c as React.ReactElement;
        return React.cloneElement(child, { disabled });
      })}
    </ButtonClusterRoot>
  );
});

ButtonCluster.defaultProps = {
  disabled: false,
  stretched: false,
};

ButtonCluster.displayName = 'ButtonCluster';

export { ButtonCluster };
