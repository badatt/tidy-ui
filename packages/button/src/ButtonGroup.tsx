import React from 'react';
import { ButtonGroupRoot } from './components';
import { IButtonGroupProps } from './types';

/**
 * ButtonGroup can be use to group several look alike buttons to form a grouped CTAs
 *
 */
const ButtonGroup = React.forwardRef<HTMLDivElement, IButtonGroupProps>((props, ref) => {
  const { children, disabled, ...rest } = props;
  return (
    <ButtonGroupRoot role="button" ref={ref} {...rest}>
      {React.Children.map(children, (c) => {
        const child = c as React.ReactElement;
        return React.cloneElement(child, { disabled });
      })}
    </ButtonGroupRoot>
  );
});

ButtonGroup.defaultProps = {
  disabled: false,
  isStretched: false,
  isUnified: false,
  isVertical: false,
};

ButtonGroup.displayName = 'ButtonGroup';

export { ButtonGroup };
