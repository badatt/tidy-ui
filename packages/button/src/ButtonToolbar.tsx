import React from 'react';
import { ButtonToolbarRoot } from './components';
import { IButtonToolbarProps } from './types';

/**
 * ButtonToolbar can group several Buttons & ButtonGroups to bring in that
 * toolbar effect, that you usually see in embedded editors
 *
 */
const ButtonToolbar = React.forwardRef<HTMLDivElement, IButtonToolbarProps>((props, ref) => {
  const { children, ...rest } = props;
  return (
    <ButtonToolbarRoot role="button" ref={ref} {...rest}>
      {children}
    </ButtonToolbarRoot>
  );
});

ButtonToolbar.defaultProps = {
  disabled: false,
};

ButtonToolbar.displayName = 'ButtonToolbar';

export { ButtonToolbar };
