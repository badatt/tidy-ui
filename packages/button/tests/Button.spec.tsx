/**
 * @jest-environment jsdom
 */
import React from 'react';
import { render } from '@testing-library/react';
import { ThemeProvider } from 'styled-components';
import '@testing-library/jest-dom';
import 'jest-styled-components';
import { AddCircleIcon } from '@tidy-ui/commons';
import { orchidDark, orchidLight } from '@tidy-ui/commons';
import { Button } from '../src';

describe('Render Button', () => {
  it('Basic render', () => {
    const tree = render(
      <ThemeProvider theme={orchidLight}>
        <Button>Basic</Button>
      </ThemeProvider>,
    );
    expect(tree).toMatchSnapshot();
  });
  it('Basic render in dark', () => {
    const tree = render(
      <ThemeProvider theme={orchidDark}>
        <Button>Basic</Button>
      </ThemeProvider>,
    );
    expect(tree).toMatchSnapshot();
  });

  it('Simple Button in dark', () => {
    const tree = render(
      <ThemeProvider theme={orchidDark}>
        <Button variant="simple">simple</Button>
      </ThemeProvider>,
    );
    expect(tree).toMatchSnapshot();
  });

  it('Outlined Button in dark', () => {
    const tree = render(
      <ThemeProvider theme={orchidDark}>
        <Button variant="outlined">outlined</Button>
      </ThemeProvider>,
    );
    expect(tree).toMatchSnapshot();
  });

  it('Hero Button in dark', () => {
    const tree = render(
      <ThemeProvider theme={orchidDark}>
        <Button variant="hero">hero</Button>
      </ThemeProvider>,
    );
    expect(tree).toMatchSnapshot();
  });

  it('Disabled button', () => {
    const tree = render(
      <ThemeProvider theme={orchidLight}>
        <Button disabled>Disabled</Button>
      </ThemeProvider>,
    );
    expect(tree).toMatchSnapshot();
  });

  it('Loading blocked button', () => {
    const tree = render(
      <ThemeProvider theme={orchidLight}>
        <Button loading>Loading...</Button>
      </ThemeProvider>,
    );
    expect(tree).toMatchSnapshot();
  });

  it('Stretch full width button', () => {
    const tree = render(
      <ThemeProvider theme={orchidLight}>
        <Button stretched>Stretched</Button>
      </ThemeProvider>,
    );
    expect(tree).toMatchSnapshot();
  });

  it('Uppercase button', () => {
    const tree = render(
      <ThemeProvider theme={orchidLight}>
        <Button uc>uppercase</Button>
      </ThemeProvider>,
    );
    expect(tree).toMatchSnapshot();
  });

  it('Button with all color variants', () => {
    const tree = render(
      <ThemeProvider theme={orchidLight}>
        <Button tone="major">major</Button>
        <Button tone="minor">minor</Button>
        <Button tone="neutral">neutral</Button>
        <Button tone="info">info</Button>
        <Button tone="success">success</Button>
        <Button tone="warning">warning</Button>
        <Button tone="danger">danger</Button>
      </ThemeProvider>,
    );
    expect(tree).toMatchSnapshot();
  });

  it('Button with all size variants', () => {
    const tree = render(
      <ThemeProvider theme={orchidLight}>
        <Button size="xxs">xxs</Button>
        <Button size="xs">size-xs</Button>
        <Button size="sm">size-sm</Button>
        <Button size="md">size-md</Button>
        <Button size="lg">size-lg</Button>
        <Button size="xl">size-xl</Button>
        <Button size="xxl">size-xxl</Button>
      </ThemeProvider>,
    );
    expect(tree).toMatchSnapshot();
  });

  it('Button with all type variants', () => {
    const tree = render(
      <ThemeProvider theme={orchidLight}>
        <Button variant="primary">primary</Button>
        <Button variant="outlined">outlined</Button>
        <Button variant="basic">basic</Button>
        <Button variant="simple">simple</Button>
        <Button variant="hero">hero</Button>
      </ThemeProvider>,
    );
    expect(tree).toMatchSnapshot();
  });
  it('Button with all type variants in dark mode', () => {
    const tree = render(
      <ThemeProvider theme={orchidDark}>
        <Button variant="primary">primary</Button>
        <Button variant="outlined">outlined</Button>
        <Button variant="basic">basic</Button>
        <Button variant="simple">simple</Button>
        <Button variant="hero">hero</Button>
      </ThemeProvider>,
    );
    expect(tree).toMatchSnapshot();
  });
  it('Button with icon', () => {
    const tree = render(
      <ThemeProvider theme={orchidLight}>
        <Button variant="primary" tone="major">
          <AddCircleIcon />
          Settings
        </Button>
      </ThemeProvider>,
    );
    expect(tree).toMatchSnapshot();
  });

  it('Button with only icon', () => {
    const tree = render(
      <ThemeProvider theme={orchidLight}>
        <Button tone="neutral">
          <AddCircleIcon />
        </Button>
      </ThemeProvider>,
    );
    expect(tree).toMatchSnapshot();
  });
});
