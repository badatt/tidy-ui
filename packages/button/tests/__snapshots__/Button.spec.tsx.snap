// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render Button Basic neutral button 1`] = `
<button
  className=""
  color="neutral"
  disabled={false}
  role="button"
  size="md"
>
  neutral
</button>
`;

exports[`Render Button Basic neutral button dark 1`] = `
<button
  className=""
  color="neutral"
  disabled={false}
  role="button"
  size="md"
>
  neutral
</button>
`;

exports[`Render Button Button with icon and text 1`] = `
<button
  className=""
  color="major"
  disabled={false}
  role="button"
  size="md"
>
  <i
    className=""
    color="major"
    loading={false}
    size="md"
  >
    <svg
      aria-hidden="true"
      fill="currentColor"
      focusable={false}
      viewBox="0 0 24 24"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm-2 15-5-5 1.41-1.41L10 14.17l7.59-7.59L19 8l-9 9z"
      />
    </svg>
  </i>
  Settings
</button>
`;

exports[`Render Button Button with icon and text dark 1`] = `
<button
  className=""
  color="major"
  disabled={false}
  role="button"
  size="md"
>
  <i
    className=""
    color="major"
    loading={false}
    size="md"
  >
    <svg
      aria-hidden="true"
      fill="currentColor"
      focusable={false}
      viewBox="0 0 24 24"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm-2 15-5-5 1.41-1.41L10 14.17l7.59-7.59L19 8l-9 9z"
      />
    </svg>
  </i>
  Settings
</button>
`;

exports[`Render Button Filled button 1`] = `
<button
  className=""
  color="major"
  disabled={false}
  role="button"
  size="md"
>
  primary
</button>
`;

exports[`Render Button Filled button dark 1`] = `
<button
  className=""
  color="major"
  disabled={false}
  role="button"
  size="md"
>
  primary
</button>
`;

exports[`Render Button Filled disabled button 1`] = `
<button
  className=""
  color="info"
  disabled={true}
  role="button"
  size="md"
>
  info
</button>
`;

exports[`Render Button Filled disabled button dark 1`] = `
<button
  className=""
  color="info"
  disabled={true}
  role="button"
  size="md"
>
  info
</button>
`;

exports[`Render Button Filled loading button 1`] = `
<button
  className=""
  color="major"
  disabled={true}
  role="button"
  size="md"
>
  <i
    className=""
    color="major"
    loading={true}
    size="md"
  >
    <svg
      enableBackground="new 0 0 0 0"
      version="1.1"
      viewBox="0 0 100 100"
      x="0px"
      xlinkHref="http://www.w3.org/1999/xlink"
      xmlSpace="preserve"
      xmlns="http://www.w3.org/2000/svg"
      y="0px"
    >
      <path
        d="M73,50c0-12.7-10.3-23-23-23S27,37.3,27,50 M30.9,50c0-10.5,8.5-19.1,19.1-19.1S69.1,39.5,69.1,50"
        fill="currentColor"
      >
        <animateTransform
          attributeName="transform"
          attributeType="XML"
          dur="1s"
          from="0 50 50"
          repeatCount="indefinite"
          to="360 50 50"
          type="rotate"
        />
      </path>
    </svg>
  </i>
  primary
</button>
`;

exports[`Render Button Filled loading button dark 1`] = `
<button
  className=""
  color="major"
  disabled={true}
  role="button"
  size="md"
>
  <i
    className=""
    color="major"
    loading={true}
    size="md"
  >
    <svg
      enableBackground="new 0 0 0 0"
      version="1.1"
      viewBox="0 0 100 100"
      x="0px"
      xlinkHref="http://www.w3.org/1999/xlink"
      xmlSpace="preserve"
      xmlns="http://www.w3.org/2000/svg"
      y="0px"
    >
      <path
        d="M73,50c0-12.7-10.3-23-23-23S27,37.3,27,50 M30.9,50c0-10.5,8.5-19.1,19.1-19.1S69.1,39.5,69.1,50"
        fill="currentColor"
      >
        <animateTransform
          attributeName="transform"
          attributeType="XML"
          dur="1s"
          from="0 50 50"
          repeatCount="indefinite"
          to="360 50 50"
          type="rotate"
        />
      </path>
    </svg>
  </i>
  primary
</button>
`;

exports[`Render Button Filled uppercase button 1`] = `
<button
  className=""
  color="danger"
  disabled={false}
  role="button"
  size="md"
>
  danger
</button>
`;

exports[`Render Button Filled uppercase button dark 1`] = `
<button
  className=""
  color="danger"
  disabled={false}
  role="button"
  size="md"
>
  danger
</button>
`;

exports[`Render Button Hero button 1`] = `
<button
  className=""
  color="minor"
  disabled={false}
  role="button"
  size="md"
>
  hero
</button>
`;

exports[`Render Button Hero button dark 1`] = `
<button
  className=""
  color="minor"
  disabled={false}
  role="button"
  size="md"
>
  hero
</button>
`;

exports[`Render Button Hero neutral button 1`] = `
<button
  className=""
  color="major"
  disabled={false}
  role="button"
  size="md"
>
  hero
</button>
`;

exports[`Render Button Hero neutral button dark 1`] = `
<button
  className=""
  color="major"
  disabled={false}
  role="button"
  size="md"
>
  hero
</button>
`;

exports[`Render Button Neutral button 1`] = `
<button
  className=""
  color="major"
  disabled={false}
  role="button"
  size="md"
>
  neutral
</button>
`;

exports[`Render Button Neutral button dark 1`] = `
<button
  className=""
  color="major"
  disabled={false}
  role="button"
  size="md"
>
  neutral
</button>
`;

exports[`Render Button Only icon button 1`] = `
<button
  className=""
  color="major"
  disabled={false}
  role="button"
  size="md"
>
  <i
    className=""
    color="major"
    loading={false}
    size="md"
  >
    <svg
      enableBackground="new 0 0 0 0"
      version="1.1"
      viewBox="0 0 100 100"
      x="0px"
      xlinkHref="http://www.w3.org/1999/xlink"
      xmlSpace="preserve"
      xmlns="http://www.w3.org/2000/svg"
      y="0px"
    >
      <path
        d="M73,50c0-12.7-10.3-23-23-23S27,37.3,27,50 M30.9,50c0-10.5,8.5-19.1,19.1-19.1S69.1,39.5,69.1,50"
        fill="currentColor"
      >
        <animateTransform
          attributeName="transform"
          attributeType="XML"
          dur="1s"
          from="0 50 50"
          repeatCount="indefinite"
          to="360 50 50"
          type="rotate"
        />
      </path>
    </svg>
  </i>
  
</button>
`;

exports[`Render Button Only icon button dark 1`] = `
<button
  className=""
  color="major"
  disabled={false}
  role="button"
  size="md"
>
  <i
    className=""
    color="major"
    loading={false}
    size="md"
  >
    <svg
      enableBackground="new 0 0 0 0"
      version="1.1"
      viewBox="0 0 100 100"
      x="0px"
      xlinkHref="http://www.w3.org/1999/xlink"
      xmlSpace="preserve"
      xmlns="http://www.w3.org/2000/svg"
      y="0px"
    >
      <path
        d="M73,50c0-12.7-10.3-23-23-23S27,37.3,27,50 M30.9,50c0-10.5,8.5-19.1,19.1-19.1S69.1,39.5,69.1,50"
        fill="currentColor"
      >
        <animateTransform
          attributeName="transform"
          attributeType="XML"
          dur="1s"
          from="0 50 50"
          repeatCount="indefinite"
          to="360 50 50"
          type="rotate"
        />
      </path>
    </svg>
  </i>
  
</button>
`;

exports[`Render Button Outlined button 1`] = `
<button
  className=""
  color="minor"
  disabled={false}
  role="button"
  size="md"
>
  secondary
</button>
`;

exports[`Render Button Outlined button dark 1`] = `
<button
  className=""
  color="minor"
  disabled={false}
  role="button"
  size="md"
>
  secondary
</button>
`;

exports[`Render Button Outlined full width button 1`] = `
<button
  className=""
  color="warning"
  disabled={false}
  role="button"
  size="md"
>
  warning
</button>
`;

exports[`Render Button Outlined full width button dark 1`] = `
<button
  className=""
  color="warning"
  disabled={false}
  role="button"
  size="md"
>
  warning
</button>
`;

exports[`Render Button Outlined neutral button 1`] = `
<button
  className=""
  color="major"
  disabled={false}
  role="button"
  size="md"
>
  secondary
</button>
`;

exports[`Render Button Outlined neutral button dark 1`] = `
<button
  className=""
  color="major"
  disabled={false}
  role="button"
  size="md"
>
  secondary
</button>
`;

exports[`Render Button Renders with no styles 1`] = `
<button
  className=""
  color="major"
  disabled={false}
  role="button"
  size="md"
>
  
</button>
`;

exports[`Render Button Renders with no styles dark 1`] = `
<button
  className=""
  color="major"
  disabled={false}
  role="button"
  size="md"
>
  
</button>
`;
